apps/glusterfs/app_block_volume.go

```diff
@@ -0,0 +1,196 @@
+//
+// Copyright (c) 2015 The heketi Authors
+//
+// This file is licensed to you under your choice of the GNU Lesser
+// General Public License, version 3 or any later version (LGPLv3 or
+// later), or the GNU General Public License, version 2 (GPLv2), in all
+// cases as published by the Free Software Foundation.
+//
+
+package glusterfs
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/boltdb/bolt"
+	"github.com/gorilla/mux"
+	"github.com/heketi/heketi/pkg/glusterfs/api"
+	"github.com/heketi/heketi/pkg/utils"
+)
+
+func (a *App) BlockVolumeCreate(w http.ResponseWriter, r *http.Request) {
+
+	var msg api.BlockVolumeCreateRequest
+	err := utils.GetJsonFromRequest(r, &msg)
+	if err != nil {
+		http.Error(w, "request unable to be parsed", 422)
+		return
+	}
+
+	if msg.Size < 1 {
+		http.Error(w, "Invalid volume size", http.StatusBadRequest)
+		logger.LogError("Invalid volume size")
+		return
+	}
+
+	if msg.Size > NewBlockHostingVolumeSize {
```

Comment:
@lpabon The NewBlockHostingVolumeSize is a configurable value, if blockvolume has to be created of bigger size then they can change the value and create the blockvolume. 

@obnoxxx Can you share your thoughts too.